libs/fetcher.client.js

```diff
@@ -49,456 +37,278 @@ function parseResponse(response) {
 }
 
 /**
- * The queue sweeps and processes items in the queue when there are items in the queue.
- * When a item is pushed into the queue, a timeout is set to guarantee the item will be processed soon.
- * If there are any item in the queue before a item, this item can be processed sooner than its timeout.
- *
- * @class Queue
+ * @class RequestClient
+ * @param {String} operation The CRUD operation name: 'create|read|update|delete'.
+ * @param {String} resource name of fetcher/service
+ * @param {Object} options configuration options for Request
  * @constructor
- * @param {String} id         ID for the queue.
- * @param {Object} config    The configuration object.
- * @param {Function} sweepFn The function to be called when queue is swept.
- * @param {Array} sweepFn.items The current items in the queue.
- * @param {Function} callback The function to be used to process a given item in the queue.
- * @param {Object} callback.item The obj that was popped from the queue.
  */
-/* istanbul ignore next */
-function Queue(id, config, sweepFn, callback) {
-    this.id = id;
-    this.config = config || {};
-    this._sweep = sweepFn;
-    this._cb = callback;
-    this._items = [];
-    this._timer = null;
+function Request (operation, resource, options) {
+    this.operation = operation;
```

Comment:
what happens if no operation or resource is passed in? should we throw or default?